crates/ide-completion/src/context.rs

```diff
@@ -155,13 +155,54 @@ pub(crate) struct ExprCtx {
 pub(crate) enum TypeLocation {
     TupleField,
     TypeAscription(TypeAscriptionTarget),
-    GenericArgList(Option<ast::GenericArgList>),
+    /// Generic argument position e.g. `Foo<$0>`
+    GenericArg(Option<(ast::GenericArgList, Option<hir::Trait>, Option<ast::GenericParam>)>),
```

Comment:
can you add a comment here for what the trait is specifically? Or alternatively turn this into a record enum with proper field names for the fields